pages/instructorFileEditor/instructorFileEditor.ejs

```diff
@@ -1,78 +1,155 @@
+                                    <span aria-hidden="true">&times;</span>
+                                </button>
+                            </div>
+                            <% if (fileEdit.failedPush) { %>
+                            <div class="row mt-4">
+                                <div class="col-auto">
+                                    In particular, it looks like there was a failure to push your changes to the remote git repo. The most likely cause is that another user made changes to other course files while you were editing. In this case, you should open the sync page in a new browser window, pull from remote git repo, close the window, and continue editing.
+                                </div>
+                            </div>
+                            <% } %>
+                            <% if (fileEdit.jobSequenceId != null) { %>
+                            <div class="row collapse mt-4" id="results-<%= fileEdit.uuid %>">
+                                <div class="card card-body">
+                                    <%- include('../partials/jobSequenceResults', {job_sequence: fileEdit.jobSequence, job_sequence_enable_live_update: false}); %>
+                                </div>
+                            </div>
+                            <% } %>
+                        </div>
+                        <% } %>
+
+                        <% if (fileEdit.alertChoice) { %>
+                        <div class="alert alert-danger alert-dismissible fade show m-2" role="alert">
+                            <% if (fileEdit.hasSameHash) { %>
+                            You were editing this file and made changes. You may choose either to continue editing or to discard your changes.
+                            <% } else { %>
+                            Both you and another user made changes to this file. You may choose either to continue editing your draft or to discard your changes.
+                            <% } %>
+                            <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                                <span aria-hidden="true">&times;</span>
+                            </button>
+                        </div>
+                        <% } %>
                     </div>
-                <% } %>
 
-                <div class="card d-flex flex-column flex-fill">
-                    <div class="card-header">
-                        <%= fileEdit.dirName %>/<%= fileEdit.fileName %>
+                    <div id="file-editor-<%= fileEdit.uuid %>-draft" class="col">
+                        <div class="card p-0">
+                            <% if (fileEdit.alertChoice) { %>
+                            <div class="card-header bg-white text-center">
+                                <button id="choose" class="btn btn-primary" type="button">Choose this version (continue editing)</button>
```

Comment:
Could we add prominent labels to the left and right sides like “My version” and “Their version”? The terminology of “my” and “theirs” is what git uses, and I think it’s fairly clear.

We could then also label the buttons “Choose my version (continue editing)” and “Choose their version (discard my changes)”.